sklearn/metrics/_classification.py

```diff
@@ -2371,6 +2371,14 @@ def hinge_loss(y_true, pred_decision, *, labels=None, sample_weight=None):
     y_true = column_or_1d(y_true)
     y_true_unique = np.unique(labels if labels is not None else y_true)
     if y_true_unique.size > 2:
+        if (pred_decision.ndim == 1) or \
+                (labels is not None and pred_decision.ndim > 1 and
+                 np.size(y_true_unique) != pred_decision.shape[1]):
+            raise ValueError("The shape of pred_decision is not "
+                             "consistent with the number of classes. "
+                             "pred_decision shape must be "
+                             "(n_samples, n_classes) with "
+                             "multiclass target")
```

Comment:
The logically flow in this section is a bit strange. I think the following would be clearer:

```python
labels = np.unique(labels if labels is not None else y_true)
if labels.size > 2:
	if pred_decision.ndim == 1 or labels.size != pred_decision.shape[1]:
		raise ValueError(...)

	le = LabelEncoder()
	le.fit(labels)
    ...
```

This would cover both cases for input validation. What do you think?

Reply:
> Thank you for the PR @PierreAttard !

It's a pleasure !

Indeed, I agreed. I made a modification that looks like what you have proposed at first. 
But in this case, we will can not cover the situation ``labels is None`` :  

```python
        if (labels is None and pred_decision.ndim > 1 and
                (np.size(y_true_unique) != pred_decision.shape[1])):
            raise ValueError("Please include all labels in y_true "
                             "or pass labels as third argument")
```

If it's ok for you, I can do the modification you proposed and adapt the below unitary test in file ``test_classification.py`` to the new situation : 

```python
def test_hinge_loss_multiclass_missing_labels_with_labels_none
```

Reply:
I am thinking of combing the two if statements into one because they almost do the same check, but looking at this more closely I think we want the two error messages. How about this:

```py
if pred_decision.ndim > 1 and np.size(y_true_unique) != pred_decision.shape[1]):
	if labels is not None:
	    raise ValueError("The shape of pred_decision is not consistent ...")
	else:
		raise ValueError("Please include ..")
```

while leaving everything else the same.

Reply:
Seems a good solution to me.

Reply:
Ah finally, we miss the situation of the intial issues #19638 : 
- ``pred_decision`` with only 1 dimension
- Whereas it is a multiclass problem

So, we can do add ```pred_decision.ndim == 1```:  
```python
if pred_decision.ndim == 1 or (pred_decision.ndim > 1 and np.size(y_true_unique) != pred_decision.shape[1]):
	if labels is not None:
	    raise ValueError("The shape of pred_decision is not consistent ...")
	else:
	    raise ValueError("Please include ..")
```

Reply:
Yea that makes sense. I think we can also remove one of the conditions:

```py
if pred_decision.ndim <= 1 or y_true_unique.size != pred_decision.shape[1]:
	if labels is not None:
	    raise ValueError("The shape of pred_decision is not consistent ...")
	else:
	    raise ValueError("Please include ..")
```

This would catch the ndim == 0 case as well. This feels like an exercise in boolean logic :)

Reply:
> This feels like an exercise in boolean logic :)

Indeed !!

Reply:
I just add ```labels is not None or pred_decision.ndim <= 1``` in order to get the right error message when you put a 1d array for ```pred_decision``` whereas it is a multiclass problem.